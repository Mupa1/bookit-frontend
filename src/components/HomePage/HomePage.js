import React from 'react';
import { Link, Redirect } from 'react-router-dom';

import styles from './HomePage.module.css';

const HomePage = () => {
  const currentUser = localStorage.getItem('token');
  if (currentUser) { return <Redirect to="/doctors" />; }
  return (
    <section className={styles.homeContainer}>
      <h1>BookIT</h1>
      <button type="button">
        <Link to="/signin" className="text-white">Sign In</Link>
      </button>
      <button type="button">
        <Link to="/signup" className="text-white">Register</Link>
      </button>
    </section>
  );
};

export default HomePage;
