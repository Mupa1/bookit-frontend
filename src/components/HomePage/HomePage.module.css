.homeContainer h1 {
  margin: 0;
  font-size: 500%;
  letter-spacing: 12px;
  padding: 100px 0;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bolder;
  text-shadow: 0 0 3px #000;
}

button {
  background-color: #96b70d;
  height: 50px;
  width: 200px;
  border-radius: 30px;
  color: #fff;
  font-weight: bold;
  font-size: 20px;
  margin: 10px;
}

button:hover {
  border: 1px solid #fff;
}

.homeContainer {
  min-height: 100vh;
  width: 100%;
  height: 100%;
  background-color: #f2b706;
  color: #fff;
  text-align: center;
  margin: auto;
  background-image: url('../../assets/nurses.png');
  background-repeat: no-repeat;
  background-position: 50% 100%;
  background-size: contain;
}

@media screen and (max-width: 320px) {
  .homeContainer h1 {
    font-size: 450%;
  }
}

@media screen and (max-width: 768px) {
  .homeContainer {
    background-size: 60vh;
  }
}
