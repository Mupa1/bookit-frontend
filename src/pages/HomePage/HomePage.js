import React from 'react';
import { Link } from 'react-router-dom';

import styles from './HomePage.module.css';

const HomePage = () => (
  <section className={styles.homeContainer}>
    <h1>BOOK AN APPOINTMENT</h1>
    <button type="button">
      <Link to="/signin" className="text-white">Sign In</Link>
    </button>
    <button type="button">
      <Link to="/signup" className="text-white">Register</Link>
    </button>
  </section>
);
export default HomePage;
