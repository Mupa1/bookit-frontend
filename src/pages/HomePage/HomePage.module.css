h1 {
  margin: 0;
  font-size: 6vw;
  letter-spacing: 12px;
  padding: 100px 0;
}

button {
  background-color: #96b70d;
  height: 50px;
  width: 200px;
  border-radius: 20px;
  color: #fff;
  font-weight: bold;
  font-size: 20px;
  margin: 10px;
}

.homeContainer {
  height: 100vh;
  width: 100%;
  background-color: #f2b706;
  color: #fff;
  text-align: center;
}
