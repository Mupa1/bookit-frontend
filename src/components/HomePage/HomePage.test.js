import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import HomePage from './HomePage';

describe('HomePage', () => {
  it('renders HomePage component with a message', () => {
    render(
      <Router>
        <HomePage />
      </Router>,
    );
    expect(screen.getByText('BookIT')).toBeInTheDocument();
  });

  it('matches Homepage snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <HomePage />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
