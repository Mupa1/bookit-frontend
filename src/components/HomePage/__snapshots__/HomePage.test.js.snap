// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches Homepage snapshot 1`] = `
<section
  className="homeContainer"
>
  <h1>
    BookIT
  </h1>
  <a
    className="text-white"
    href="/signin"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      Sign In
    </button>
  </a>
  <a
    className="text-white"
    href="/signup"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      Register
    </button>
  </a>
</section>
`;
